asserts/batch.go

```diff
@@ -217,7 +217,7 @@ func (b *Batch) prereqSort(db *Database) error {
 		ordered = append(ordered, a)
 		return nil
 	}
-	f := NewFetcher(db, retrieve, save)
+	f := NewFetcher(db, retrieve, nil, save)
```

Comment:
I'm not super happy about this signature change as now there is a `nil` and it's hard to see what that means and also a fetcher without a retrieveSeq callback would crash (easy enough to fix of course).

So I played around a bit and was wondering if something like adding a new interface `SeqFetcher` that embedds `Fetcher` would make things a bit nicer. Full diff of my idea here: https://github.com/snapcore/snapd/compare/master...mvo5:fetcher-sequenced-alternative-idea?expand=1

So while the (unexported) fetcher implements both the Fetcher and SeqFetcher interface by returning `NewFetcher() Fetcher`  and `NewSeqFetche() SeqFetcher` interfaces we ensure that a fetcher without a retrieveSeq cannot be used to call FetchSequence.

Reply:
I really love this suggestion, I've cherry-picked your commit